feature/gribi/ate_tests/route_removal_non_default_vrf_test/route_removal_non_default_vrf_test.go

```diff
@@ -0,0 +1,454 @@
+func injectEntries(ctx context.Context, t *testing.T, dut *ondatra.DUTDevice, client *gribi.Client, networkInstanceName string, ateDstNetCIDR string) {
+	t.Logf("Add an IPv4Entry for %s pointing to ATE port-2 via gRIBI client", ateDstNetCIDR)
+	client.AddNH(t, nhIndex, atePort2.IPv4, *deviations.DefaultNetworkInstance, fluent.InstalledInRIB)
+	client.AddNHG(t, nhgIndex, map[uint64]uint64{nhIndex: 1}, *deviations.DefaultNetworkInstance, fluent.InstalledInRIB)
+	client.AddIPv4(t, ateDstNetCIDR, nhgIndex, networkInstanceName, *deviations.DefaultNetworkInstance, fluent.InstalledInRIB)
+}
+
+// injectIPEntry adds only IPv4 entry to the specified network instance referencing to the nhgid, to the VRF.
+func injectIPEntry(ctx context.Context, t *testing.T, dut *ondatra.DUTDevice, client *gribi.Client, networkInstanceName string, ateDstNetCIDR string) {
+	t.Logf("Add an IPv4Entry for %s via gRIBI client's %s network instance", ateDstNetCIDR, networkInstanceName)
+	client.AddIPv4(t, ateDstNetCIDR, nhgIndex, networkInstanceName, *deviations.DefaultNetworkInstance, fluent.InstalledInRIB)
+
+	// After adding the entry, verify the entry is active through AFT Telemetry.
+	ipv4Path := dut.Telemetry().NetworkInstance(networkInstanceName).Afts().Ipv4Entry(ateDstNetCIDR)
+	if got, ok := ipv4Path.Prefix().Watch(t, time.Minute, func(val *telemetry.QualifiedString) bool {
+		return val.IsPresent() && val.Val(t) == ateDstNetCIDR
+	}).Await(t); !ok {
+		t.Errorf("ipv4-entry/state/prefix got %s, want %s", got.Val(t), ateDstNetCIDR)
+	}
+}
+
+// validateNotPrimaryError validates the canonical and exact error details for flush operation of non-primary client.
+func validateNotPrimaryError(t *testing.T, dut *ondatra.DUTDevice, flushErr error) {
+	s, ok := status.FromError(flushErr)
+	if !ok {
+		t.Fatalf("received invalid error from server, got: %v", flushErr)
+	}
+	if got, want := s.Code(), codes.FailedPrecondition; got != want {
+		t.Fatalf("did not get the expected canonical error code from server, got code: %s (error: %v), want: %s", got, flushErr, want)
+	}
+	if len(s.Details()) != 1 {
+		t.Fatalf("got more than 1 error details, got: %v", flushErr)
+	}
+	gotD, ok := s.Details()[0].(*gpb.FlushResponseError)
+	if !ok {
+		t.Fatalf("did not get the expected error details type, got: %T, want: *gpb.FlushResponseError", s.Details()[0])
+	}
+	wantD := &gpb.FlushResponseError{
+		Status: gpb.FlushResponseError_NOT_PRIMARY,
+	}
+	if !proto.Equal(gotD, wantD) {
+		t.Fatalf("did not get the exact error details, got: %s, want: %s", prototext.Format(gotD), prototext.Format(wantD))
+	}
+}
+
+// flush function flushes all the state on the server, but does not validate it specifically.
+func flush(ctx context.Context, t *testing.T, client *fluent.GRIBIClient, electionID uint64, networkInstanceName string) (*gpb.FlushResponse, error) {
+	t.Helper()
+	res, err := client.Flush().
+		WithElectionID(electionID, 0).
+		WithNetworkInstance(networkInstanceName).
+		Send()
+	return res, err
```

Comment:
This can be merged into a single statement
```
	return client.Flush().
		WithElectionID(electionID, 0).
		WithNetworkInstance(networkInstanceName).
		Send()
```